### PR TITLE
Cypress: re-organize tests and CI

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé

This PR reorganizes the cypress test suite to run faster and be clearer to understand.  The refactor breaks out the 4 a11y tests we currently do into re-usable features:
1. Validates a11y rules using AXE Dequeue
2. Validates HTML using `html-validate`
3. Checks for dead-links
4. Checks that any referenced svgs in a page has the correct mime-type

It also removes testing of the GCA pages from the PR review workflow, since failures here don't have anything to do with the PR.  Instead, a scheduled cypress run has been added against staging every 3 hours.  This run will check all the app pages and GCA pages.

### CODE REVIEW

File: tests_cypress/cypress/support/commands.js [hunk 1]
```diff
@@ -22,4 +22,47 @@
...
+        cy.get('body').within(() => {
+            cy.get('a').each((link) => {
+                if (link.prop('href').startsWith('mailto') || link.prop('href').startsWith('/set-lang') || link.prop('href').startsWith('#')) return;
+                cy.request(link.prop('href'));
+            });
+        });
+    }
+
+    // 4. check for correct mime type on svg images (should be image/svg+xml)
+    if (options.mimeTypes) {
```
Comment (jzbahrai) [inline tests_cypress/cypress/support/commands.js:58]: very nice

**Review by jzbahrai — APPROVED** [review]

File: .github/workflows/cypress-staging.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Cypress staging a11y tests
```
Comment (andrewleith) [inline .github/workflows/cypress-staging.yaml:1]: New scheduled workflow to run a11y tests against staging every 3 hours

File: tests_cypress/cypress/e2e/admin/a11y/app_pages.cy.js [hunk 3]
```diff
@@ -31,37 +32,15 @@ const pages = [
...
-                            });
-                        });
-                    });
-                }
-    //         });
-    //     })
-    // }
-});
\ No newline at end of file
+    for (const page of pages) {
```
Comment (andrewleith) [inline tests_cypress/cypress/e2e/admin/a11y/app_pages.cy.js:41]: For the app pages, don't bother checking for dead links (we rarely if ever link outside of the app) or for mime-types.

File: tests_cypress/cypress/e2e/admin/a11y/gca_pages.cy.js [hunk 4]
```diff
@@ -26,55 +26,15 @@ const fullPageList = [
...
-                    });
-                }
-            });
+describe(`GCA a11y tests [${config.CONFIG_NAME}]`, () => {
+    for (const lang of langs) {
+        const currentLang = (lang === 'en' ? 'English' : 'Francais');
+        context(currentLang, () => {
+            for (const page of fullPageList) {
+                it(`${page[lang]}`, () => {
+                    cy.a11yScan(page[lang]);
```
Comment (andrewleith) [inline tests_cypress/cypress/e2e/admin/a11y/gca_pages.cy.js:35]: Scan for everything for the GCA pages (a11y, html validity, dead links, and svg mime-types)

File: tests_cypress/cypress/support/commands.js [hunk 5]
```diff
@@ -22,4 +24,94 @@
 //
 //
 // -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
\ No newline at end of file
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
```
Comment (andrewleith) [inline tests_cypress/cypress/support/commands.js:27]: Update logic here to never check the same link or the same svg twice.  This makes the tests much faster and reduces the number of network requests we make to Notify.

**Review by jzbahrai — APPROVED** [review]

**Pull request merged**